### PR TITLE
Added deprecation warning to Array.groupByKey(keyForValue:) method

### PR DESCRIPTION
**Pull request by calebkleveter**
## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [ ] New extensions are written in Swift 4.
- [ ] New extensions support iOS 8.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+.
- [ ] I have added tests for new extensions, and they passed.
- [ ] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [ ] All extensions are declared as **public**.
- [x] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

### CODE REVIEW

**Comment by calebkleveter** [comment]
@LucianoPAlmeida Fixed.

File: Sources/Extensions/SwiftStdlib/Deprecated/SwiftStdlibDeprecated.swift [hunk 1]
```diff
@@ -122,6 +122,21 @@ public extension Array {
 		return self[index]
 	}
 
+    /// SwifterSwift: Group the elements of the array in a dictionary.
+    ///
+    ///     [0, 2, 5, 4, 7].groupByKey { $0%2 ? "evens" : "odds" } -> [ "evens" : [0, 2, 4], "odds" : [5, 7] ]
+    ///
+    /// - Parameter getKey: Clousure to define the key for each element.
+    /// - Returns: A dictionary with values grouped with keys.
+    @available(*, deprecated, message: "Use 'Dictionary.init(grouping:by:)' instead.")
```
Comment (SD10) [inline Sources/Extensions/SwiftStdlib/Deprecated/SwiftStdlibDeprecated.swift]: Think this needs an argument after deprecated. `deprecated: 4.4`

Comment (SD10) [reply Sources/Extensions/SwiftStdlib/Deprecated/SwiftStdlibDeprecated.swift]: nevermind. apparently this passed CI 😱

**Comment by comm1x** [comment]
Why did you deprecate this method? I try to find any info about this, but find nothing.

**Comment by LucianoPAlmeida** [comment]
@comm1x We deprecated it right after swift standard library introduced [init(grouping: )](https://developer.apple.com/documentation/swift/dictionary/3127163-init)

**Pull request merged**